tests/python/pants_test/pants_run_integration_test.py

```diff
@@ -284,6 +299,13 @@ def run_pants_with_workdir_without_waiting(self, command, workdir, config=None,
                    '--cache-bootstrap-read', '--cache-bootstrap-write'
                    ])
 
+    if self.should_configure_pantsd():
+      args.append("--enable-pantsd=True")
+      args.append("--no-shutdown-pantsd-after-run")
+    else:
+      # NB. By unsetting this explicitly we avoid PANTS_ENABLE_PANTSD leaking into the test.
```

Comment:
Re: this `else`: this is the effect of hermetic integration tests... it might be good not to have two facilities for filtering the environment.

Reply:
Yeah, this is fair :) If it starts to become troublesome when we enable pantsd by default, we can try to make the tests hermetic.

Reply:
Opened #7525 to track